tests/test_pipenv.py

```diff
@@ -820,6 +820,27 @@ def test_complex_deps_lock_and_install_properly(self):
             c = p.pipenv('install')
             assert c.return_code == 0
 
+    @pytest.mark.lock
+    @pytest.mark.requirements
+    @pytest.mark.complex
+    def test_complex_lock_changing_candidate(self):
+        # The requests candidate will change from latest to <2.12.
```

Comment:
Do we want to check the version numbers in the lock file to see if it reports <2.12

Reply:
That would be good idea to make sure that we're resolving *and* giving a valid result, indeed!

Reply:
Updated the test